<!DOCTYPE html>
<html>
    <head>
        <meta charset"utf-8">
        <title>Arya Kaul | Projects</title>
        <meta name="description" content="Arya Kaul's Projects"> 
	    <link rel="stylesheet" href="../css/styles.css"> 
        <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.2.0/css/all.css" integrity="sha384-hWVjflwFxL6sNzntih27bfxkr27PmbbK/iSvJ+a4+0owXq79v+lsFkW54bOGbiDQ" crossorigin="anonymous">
    </head>
    
    <body>
    
    <header class="site-header">
    
    <div id="container">
	    <p class="site-title">Arya Kaul</p>
		<div id="menu">
		    <ul id="nav">
			<li><a href="../index.html">About</a></li>
			<li><a href="../publication-page/pub.html"> Publications </a></li>
			<li><a class="active" href="../projects-page/projects.html">Projects</a></li>
			<!--<li><a href="./blog-page/blog.html">Blog</a></li>-->
			<li><a href="../resume-page/res.html">Resume</a></li>
			<li><a href="../contact-page/contact.html">Contact</a></li>
		    </ul>
		</div><br>
        
        <!--SIMDIGREE-->
        <div id="project-wrapper" style="background-color:TODO">
            <div class="project-left-odd">
                <p id="name"><img src="../pics/simdigree.png" width=110 hspace="20" vspace="10"> </img> 
                Simdigree</p>
                <span id="authors">Arya Kaul, Onuralp Soylemez, Shamil Sunyaev</span>
            </div> 

            <div class="project-right-odd"> 
                <p id="description"> Simdigree is a tool to simulate pedigrees using the VCF files from runs of SLiM, a forward time population genetics simulator. The purpose of the project is to attempt to quantify the probability a given rare or undiagnosed disease is monogenic, oligogenic, or polygenic given family history. The project is still under development; however, all code is freely available on Github, and when the project is completed will be uploaded to bioconda.</p>
                    <ul style="text-align:center">
                    <a href="https://github.com/aryakaul/simdigree" target="_blank"><li class="fab fa-fw fa-3x fa-github"></li></a> 
                    </ul>
            </div>
        </div> 
  
        <!--NovaSplice-->
        <div id="project-wrapper">
            <div class="project-left-even">
                <p id="description">NovaSplice is a command line tool to predict novel splicing events due to single nucleotide polymorphisms within non-coding regions. The tool uses 3 inputs, (1) a reference genome, (2) a genome annotation file (GTF) and (3) a VCF file. NovaSplice then loops through every variant in the VCF file and computes the score of the canonical splice sites closest to that variant using maxentscan. Next, NovaSplice computes the score of every splice site containing that variant and compares it to the canonical splice site's score. If the novel splice site's score is close to or greater than the canonical splice site, it is reported. Finally, NovaSplice outputs an ordered list of predicted novel splice sites. All code is available and Github, and an executable command line version of the tool may be installed through bioconda.</p>
                    <ul style="text-align:center">
                    <a href="https://bioconda.github.io/recipes/novasplice/README.html" target="_blank"><li class="fab fa-fw fa-3x fa-cuttlefish"></li></a>
                    <a href="https://github.com/aryakaul/novasplice" target="_blank"><li class="fab fa-fw fa-3x fa-github"></li></a> 
                    </ul>
            </div> 

            <div class="project-right-even"> 
                <p id="name"><img src="../pics/novasplice.png" width=140 hspace="10" vspace="10"></img><a href="https://aryakaul.github.io/novasplice" target="_blank">NovaSplice</a></p>
                <span id="authors">Arya Kaul, Shamil Sunyaev</span>
            </div>
        </div> 
        
        
        
        <!--FIT-HI-C -->
        <div id="project-wrapper" style="background-color:TODO">
            <div class="project-left-odd">
                <p id="name"><img src="../pics/fit-hi-c.png" width=110 hspace="20" vspace="10"> </img> 
                <a href="http://ay-lab.github.io/fithic" target="_blank"> Fit-Hi-C </a></p>
                <span id="authors">Ferhat Ay, Arya Kaul</span>
            </div> 

            <div class="project-right-odd"> 
                <p id="description">Fit-Hi-C is a Python3 tool to generate statistical confidence estimates for Hi-C data to reveal significant chromatin contacts. Fit-Hi-C works by fitting a monotonically decreasing spline to the probability of a contact given the genomic distance of the interaction genetic loci. The null hypothesis may then be refined by refitting a new spline after discarding outliers. Fit-Hi-C was first developed by Dr. Ferhat Ay, and I've been working on optimizing the code for speed and memory usage. Fit-Hi-C is freely available and easily installable through the Python Package Index, Github, or Bioconda.</p>
                    <ul style="text-align:center">
                    <a href="https://pypi.org/project/fithic/" target="_blank"><li class="fab fa-fw fa-3x fa-python"></li></a> 
                    <a href="https://github.com/ay-lab/fithic" target="_blank"><li class="fab fa-fw fa-3x fa-github"></li></a> 
                    <a href="https://bioconda.github.io/recipes/fithic/README.html" target="_blank"><li class="fab fa-fw fa-3x fa-cuttlefish"></li></a>
                    </ul>
            </div>
        </div> 


        <!--HICKRY -->
        <div id="project-wrapper">
            <div class="project-left-even">
                <p id="description">HiCKRy is a Python3 tool to normalize Hi-C contact maps using the Knight-Ruiz algorithm for matrix balancing. The normalization is done under an assumption of equal visibility i.e. if there were no bias then each locus in the genome would give rise to an equal number of reads in a Hi-C experiment. This normalization allows the user to correct for all factors that may cause bias without explicitly modeling them. The KR algorithm has been demonstrated to be significantly faster than the iterative method employed by Imakaev et al., and has been shown to be scalable to high resolution Hi-C datasets. HiCKRy was entirely coded by myself, with guidance and support from Dr. Ferhat Ay. HiCKRy is freely available <!--and easily installable through the Python Package Index--> on Github.</p>
                    <ul style="text-align:center">
                    <!--<a href="https://pypi.org/project/hickry/" target="_blank"><li class="fa fa-fw fa-3x fa-pied-piper-pp"></li></a>-->
                    <a href="https://github.com/ay-lab/HiCKRy" target="_blank"><li class="fab fa-fw fa-3x fa-github"></li></a> 
                    </ul>
            </div> 

            <div class="project-right-even"> 
                <p id="name"><img src="../pics/hickry.png" width=140 hspace="10" vspace="10"></img>HiCKRy</p>
                <span id="authors">Arya Kaul, Ferhat Ay</span>
            </div>
        </div> 



        <!--HOTWASP-->
        <div id="project-wrapper">

            <div class="project-left-odd"> 
                <p id="name"><img src="../pics/hotwasp.png" width=100 hspace="20" vspace="10"> </img> 
                HotWASp</p>
                <span id="authors">Arya Kaul, <br>Alex Hakansson, Vincent Hu</span>
            </div>
            <div class="project-right-odd">
                <p id="description">Genome wide association studies (GWAS) represent a powerful tool to interrogat the relationship between genetic variation and diseases. Unfortunately, conventional GWAS studies suffer from an extremely high statistical burden due to the extreme number of hypotheses being tested. To help boost the statistical power of GWAS, HotWASp was developed. HotWASp works by running heat propagation on a given gene-gene network with initial heats inversely correlated with the p-values found through GWAS. HotWASp was developed as part of the final project for BNFO 286, a graduate level course on Network Biology and Biomedicine taught by Dr. Trey Ideker. Code is freely accessible through Github.</p>
                    <ul style="text-align:center">
                    <a href="https://github.com/aryakaul/HotWASp" target="_blank"><li class="fab fa-fw fa-3x fa-github"></li></a> 
                    </ul>
            </div> 
        </div> 

        <!--TEREX-->
        <div id="project-wrapper">
            <div class="project-left-even">
                <p id="description">Transposable elements (TEs) are unique DNA sequences capable of autonomously 'moving' around the genome. Found abundantly throughout eukaryotic and prokaryotic genomes, TEs are capable of driving genomic evolution and gene regulation by enabling translocation, causing exon shuffling, and increasing genetic diversity. TErex presents a simple tool to easily identify all transposable elements given an input fasta sequence file using the Dfam database. TErex was developed as part of the final project for CSE 180, Biology meets Computing. The tool is freely accessible through Github.</p>
                    <ul style="text-align:center">
                    <a href="https://github.com/aryakaul/TErex1.0" target="_blank"><li class="fab fa-fw fa-3x fa-github"></li></a> 
                    </ul>
            </div> 

            <div class="project-right-even"> 
                <p id="name"><img src="../pics/terex.png" width=100 hspace="20" vspace="10"> </img> 
                TErex</p>
                <span id="authors">Arya Kaul, Jenny Lee, Lowan Kim, Jigar Patel</span>
            </div>
        </div> 
    </div>
    <div class="site-footer">
         <div class="footer-content">
         <div id="left"><p>Arya Kaul<br>a1kaul@ucsd.edu<br></p></div>
         <div id="right"><p>Department of Bioengineering<br>UC San Diego<br>9500 Gilman Drive<br>La Jolla, CA 92093-0407</p></div>
         </div>
    </div>


    </body>
</html>

